### PR TITLE
Week2 comments section

### PR DESCRIPTION
**Pull request by jainsana21**
Step 4 and 5 of week 2 walkthrough. The comment section is all done now!

### CODE REVIEW

File: portfolio/pom.xml [hunk 1]
```diff
@@ -41,4 +55,7 @@
       </plugin>
     </plugins>
   </build>
-</project>
\ No newline at end of file
+</project>
```
Comment (emwarman) [inline portfolio/pom.xml:58]: What do you think about removing this extra whitespace?

Comment (emwarman) [inline portfolio/pom.xml:58]: What do you think about removing this extra whitespace?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,19 +14,72 @@
...
+        messages = new ArrayList<>();
+    }
+  
+  @Override 
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("text-input");
+    long time = System.currentTimeMillis(); 
+    messages.add(text);
+    System.out.println(messages);
```
Comment (emwarman) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:50]: Is this a debug statement? What do you think about removing it?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,19 +14,72 @@
...
+      long id = entity.getKey().getId();
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("time");
+    }
+  }
+
+  private String convertToJsonUsingGson(ArrayList data) {
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+    return json;
```
Comment (emwarman) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:83]: What do you think about condensing these last 2 lines into:

return gson.toJson(messages);

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -27,8 +27,10 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-function getData() {
-  fetch('/data').then(response => response.text()).then((quote) => {
-    document.getElementById('data-container').innerText = quote;
-  });
+async function getData() {
+  const response = await fetch('/data');
+  const Data = await response.json();
```
Comment (emwarman) [inline portfolio/src/main/webapp/script.js:32]: just curious, why is Data capitalized?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -14,19 +14,72 @@
...
+    String json = convertToJsonUsingGson(messages);
+
+    //Send response as JSON
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+    Query query = new Query("text").addSort("time", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); 
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> messages = new ArrayList<>();
```
Comment (emwarman) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:72]: Should this be at the top of the method? it looks like line 63 uses the messages variable.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -14,19 +14,72 @@
...
+    Query query = new Query("text").addSort("time", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); 
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> messages = new ArrayList<>(); 
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("time");
+    }
```
Comment (emwarman) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:77]: Where are you adding the entities to the messages list?

File: portfolio/pom.xml (hunk 1 above)
Comment (jainsana21) [reply portfolio/pom.xml:54]: Should be all cleaned up now!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (jainsana21) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:83]: Done!

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (jainsana21) [reply portfolio/src/main/webapp/script.js:32]: I believe one of the examples had it that way, so I just kind of followed their format. No particular reason though.

File: portfolio/pom.xml (hunk 1 above)
Comment (jainsana21) [reply portfolio/pom.xml:54]: Done!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (jainsana21) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:50]: Removed the system.out.println(messages) statement.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (jainsana21) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:72]: Removed this line, there was really no need for it here again. Thanks for pointing that out :)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (jainsana21) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:77]: Should be fixed now.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (emwarman) [reply portfolio/src/main/webapp/script.js:32]: ok, thanks!

typically we reserve capital variables for class/library names. We use camelCase for variable names.

**Comment by emwarman** [comment]
Looks great! Nice job!

**Pull request merged**